Modules/Core/Common/include/itkMacro.h

```diff
@@ -1373,6 +1373,14 @@ itkDynamicCastInDebugMode(TSource x)
 #endif
 }
 
+// ITK_FUTURE_DEPRECATED is only for internal use, within the implementation of ITK. It allows triggering "deprecated"
+// warnings when legacy support is removed, which warn that a specific feature may be removed in the future.
+#ifdef ITK_LEGACY_REMOVE
```

Comment:
I think this may need to be "ITK_LEGACY_REMOVE && !ITK_LEGACY_SILENT", so that test builds can be silenced.